cli/src/pcluster/config/update_policy.py

```diff
@@ -153,8 +175,60 @@ def __eq__(self, other):
     >= (change.old_value if change.old_value is not None else DEFAULT_MAX_COUNT),
 )
 
+
+def fail_reason_queue_update_strategy(change, _):
+    reason = "All compute nodes must be stopped"
+    # QueueUpdateStrategy can override UpdatePolicy of parameters under SlurmQueues
+    if any(path.startswith("SlurmQueues[") for path in change.path):
+        reason += " or queue update strategy must be set"
+
+    return reason
+
+
+def condition_checker_queue_update_strategy(change, patch):
+    result = not patch.cluster.has_running_capacity()
+    # QueueUpdateStrategy can override UpdatePolicy of parameters under SlurmQueues
+    if any(path.startswith("SlurmQueues[") for path in change.path):
+        result = result or (
+            patch.target_config.get("Scheduling").get("SlurmSettings", {}).get("QueueUpdateStrategy") is not None
+            and patch.target_config.get("Scheduling").get("SlurmSettings", {}).get("QueueUpdateStrategy")
+            != QueueUpdateStrategy.COMPUTE_FLEET_STOP.value
+        )
+
+    return result
+
+
+# Update supported only with all compute nodes down or with replacement policy set different from COMPUTE_FLEET_STOP
+UpdatePolicy.QUEUE_UPDATE_STRATEGY = UpdatePolicy(
+    name="QUEUE_UPDATE_STRATEGY",
+    level=5,
+    fail_reason=fail_reason_queue_update_strategy,
+    action_needed=UpdatePolicy.ACTIONS_NEEDED["pcluster_stop_conditional"],
+    condition_checker=condition_checker_queue_update_strategy,
+)
+
+
+def condition_checker_fleet_stop_on_remove(change, patch):
+    result = not patch.cluster.has_running_capacity()
+    # SlurmQueue or ComputeResource can be added but removal require compute fleet stop
+    if change.is_list and (any(path.startswith("SlurmQueues[") for path in change.path) or change.key == "SlurmQueues"):
```

Comment:
This test `any(path.startswith("SlurmQueues[") for path in change.path)` is replicated in multiple places we can define a function for this.

Reply:
done